pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1349,10 +1473,81 @@ func setTerminationReasonByPhase(phase v1alpha1.MachinePhase, terminationConditi
 	}
 }
 
+// TryLock tries to write to channel. It times out after specified duration
+func TryLock(lockC chan<- struct{}, duration time.Duration) bool {
+	ctx, cancelFn := context.WithTimeout(context.Background(), duration)
+	defer cancelFn()
+	for {
+		select {
+		case lockC <- struct{}{}:
+			return true
+		case <-ctx.Done():
+			return false
+		}
+	}
+}
+
+func (c *controller) tryMarkingMachineFailed(ctx context.Context, machine, clone *v1alpha1.Machine, machineDeployName, description string, lockAcquireTimeout time.Duration) (machineutils.RetryPeriod, error) {
+	defer c.permitGiver.ReleasePermit(machineDeployName)
```

Comment:
Shouldn't this be only done if you were able to acquire a permit? So essentially move this inside the if condition after you are able to acquire a permit

Reply:
I was of the opinion that anyways `ReleasePermit` has the logic to deal with permit not acquired case, but yes moving it inside would be better.

Reply:
addressed in 
d4e34d8